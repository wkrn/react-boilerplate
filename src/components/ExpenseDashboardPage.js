import React from 'react';
const ExpenseDashboardPage = () => (
    <div>
        This is from my dashboards
    </div>
)

export default ExpenseDashboardPage