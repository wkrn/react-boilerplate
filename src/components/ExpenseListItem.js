import moment from 'moment';
import React from 'react';
import { Link } from 'react-router-dom';
import numeral from 'numeral';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <Link to={`/edit/${id}`}>
        <div className="list-item">
            <h3 className="list-item__title">{description}</h3>
            <span className="list-item__subtitle">{moment(createdAt).format("YYYY-MM-DD")}</span>
            <h3 className="list-item__data">{numeral(amount / 100).format('$0.0.00')}</h3>
        </div>

    </Link>
)

export default ExpenseListItem