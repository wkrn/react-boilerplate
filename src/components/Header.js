import React from 'react';
import { NavLink } from 'react-router-dom';
const Header = () => (
    <header>
        <h2>Expensify</h2>
        <NavLink to="/" activeClassName="is-active" exact={true}>DashBoard </NavLink>
        <NavLink to="/create" activeClassName="is-active">Create </NavLink>
        <NavLink to="/edit" activeClassName="is-active">Edit </NavLink>
        <NavLink to="/help" activeClassName="is-active">Help </NavLink>
    </header>
)

export default Header