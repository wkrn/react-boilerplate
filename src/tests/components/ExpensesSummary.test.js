import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { ExpensesSummary } from '../../components/ExpensesSummary';


test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={695} />)
    expect(wrapper).toMatchSnapshot();
})

test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={73738383} />)
    expect(wrapper).toMatchSnapshot();
})