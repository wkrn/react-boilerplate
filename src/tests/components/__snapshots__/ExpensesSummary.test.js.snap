// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $6.95
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $737,383.83
  </h1>
</div>
`;
